components/match2/wikis/leagueoflegends/big_match.lua

```diff
@@ -0,0 +1,345 @@
+---
+-- @Liquipedia
+-- wiki=leagueoflegends
+-- page=Module:BigMatch
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Arguments = require('Module:Arguments')
+local Array = require('Module:Array')
+local HeroIcon = require('Module:ChampionIcon')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local MatchLinks = mw.loadData('Module:MatchLinks')
+local Match = require('Module:Match')
+local Operator = require('Module:Operator')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Tabs = require('Module:Tabs')
+local TemplateEngine = require('Module:TemplateEngine')
+local VodLink = require('Module:VodLink')
+
+local CustomMatchGroupInput = Lua.import('Module:MatchGroup/Input/Custom', {requireDevIfEnabled = true})
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper', {requireDevIfEnabled = true})
+local Template = Lua.import('Module:BigMatch/Template', {requireDevIfEnabled = true})
+local WikiSpecific = Lua.import('Module:Brkts/WikiSpecific', {requireDevIfEnabled = true})
+
+local BigMatch = {}
+
+local KEYSTONES = Table.map({
+	-- Precision
+	'Press the Attack',
+	'Lethal Tempo',
+	'Fleet Footwork',
+	'Conqueror',
+
+	-- Domination
+	'Electrocute',
+	'Predator',
+	'Dark Harvest',
+	'Hail of Blades',
+
+	-- Sorcery
+	'Summon Aery',
+	'Arcane Comet',
+	'Phase Rush',
+
+	-- Resolve
+	'Grasp of the Undying',
+	'Aftershock',
+	'Guardian',
+
+	-- Inspiration
+	'Glacial Augment',
+	'Unsealed Spellbook',
+	'First Strike',
+}, function (_, value)
+	return value, true
+end)
+
+local ROLE_ORDER = Table.map({
+	'top',
+	'jungle',
+	'middle',
+	'bottom',
+	'support',
+}, function (key, value)
+	return value, key
+end)
+
+local NOT_PLAYED = 'np'
+local DEFAULT_ITEM = 'EmptyIcon'
+local TEAMS = Array.range(1, 2)
+
+function BigMatch.run(frame)
+	local args = Arguments.getArgs(frame)
+
+	args = BigMatch._contextualEnrichment(args)
+
+	local model = BigMatch._match2Director(args)
+
+	model.isBestOfOne = #model.games == 1
+	model.dateCountdown = DisplayHelper.MatchCountdownBlock(model)
+
+	-- Create an object array for links
+	model.links = Array.extractValues(Table.map(model.links, function (site, link)
+		return site, Table.mergeInto({link = link}, MatchLinks[site])
+	end))
+
+	-- Add more opponent data field
+	Array.forEach(model.opponents, function (opponent, index)
+		opponent.opponentIndex = index
+		opponent.iconDisplay = mw.ext.TeamTemplate.teamicon(opponent.template)
+		opponent.shortname = mw.ext.TeamTemplate.raw(opponent.template).shortname
+		opponent.page = mw.ext.TeamTemplate.raw(opponent.template).page
+		opponent.name = mw.ext.TeamTemplate.raw(opponent.template).name
```

Comment:
wouldn't it be more efficient if we only access `mw.ext.TeamTemplate.raw` once instead of 3 times?
or is it cached?

```lua
		local teamTemplateData = mw.ext.TeamTemplate.raw(opponent.template)
		opponent.shortname = teamTemplateData.shortname
		opponent.page = teamTemplateData.page
		opponent.name = teamTemplateData.name
```

Reply:
It's cached